src/Domain/Creators/BinaryCompactObjectHelpers.hpp

```diff
@@ -151,6 +211,57 @@ struct TimeDependentMapOptions {
   create_functions_of_time(const std::unordered_map<std::string, double>&
                                initial_expiration_times) const;
 
+  /*!
+   * \brief Construct the actual maps that will be used.
+   *
+   * Currently, this constructs a:
+   *
+   * - Expansion: `CubicScale<3>`
+   * - Rotation: `Rotation<3>`
+   * - SizeA/B: `SphericalCompression`
+   *
+   * If the inner/outer radii for an object are `std::nullopt`, this means that
+   * a Size map is not constructed for that object. An identity map will be used
+   * instead.
+   */
+  void build_maps(
```

Comment:
If I understand correctly this function constructs the maps, and the other functions copy and return them. That's a bit of an awkward interface. Just have a private function that constructs each map with the needed arguments (like x_coord), then use those to build the compositions (passing args like x_coord along). There's no benefit in constructing-then-copying maps vs constructing on demand (that I can see).

Reply:
The reason I chose this interface was because I didn't want to have to pass the same arguments (xcoord, ...) to the three different functions that build the maps. Visually in the domain creators that would just look cluttered, so I opted to just pass the arguments into a single function and build all the maps at once. Then when you need the composition, the function you call has 0 or 1 arguments (the book).

This also makes adding maps in the future even easier because there's only one place you change arguments rather than like 6.

Reply:
I'm not convinced this changing-state approach is better, but since I won't be able to continue this review until Thursday I'm happy to step back and have another person take over this review. If none can be found I can resume the review later this week of course.

Reply:
I don't really understand the exact concern with changing state. Doing so in the constructor of the domain creator seems like a quite reasonable thing to do since setting the state of everything is the job of the constructor. I agree with Kyle that the shorter function calls and not needing to either compute more in `create_domain()` or store the inputs to the `build_maps` function as members of the domain creator is really nice. Given the current discussion here, I'm in favor of the current design Kyle has as opposed to the suggested change.